[Safer CPP] Address UnretainedLocalVars warnings in WKPDFHUDView.mm

aprotyas opened the pull request:
#### 8c383bacdec0121e9731ef9f3d75a6b814336fb7
<pre>
[Safer CPP] Address UnretainedLocalVars warnings in WKPDFHUDView.mm
<a href="https://bugs.webkit.org/show_bug.cgi?id=291077">https://bugs.webkit.org/show_bug.cgi?id=291077</a>
<a href="https://rdar.apple.com/148588361">rdar://148588361</a>

Reviewed by Megan Gardner, Chris Dumez, and Wenson Hsieh.

This patch addresses the aforementioned class of Safer CPP warnings by
introducing more RetainPtr adoption within WKPDFHUDView.mm.

* Source/WebKit/SaferCPPExpectations/UnretainedLocalVarsCheckerExpectations:
* Source/WebKit/UIProcess/PDF/WKPDFHUDView.mm:
(-[WKPDFHUDView handleMouseUp:]):
(-[WKPDFHUDView _controlIndexForEvent:]):
(-[WKPDFHUDView _loadIconImages]):
(-[WKPDFHUDView _setupLayer:]):
(-[WKPDFHUDView _redrawLayer]):
(-[WKPDFHUDView _imageForControlName:]):
(-[WKPDFHUDView _getImageForControlName:]): Deleted.

Drive-by fix: Renamed since it does not return through out arguments --
align with <a href="https://webkit.org/code-style-guidelines/#names-out-argument.">https://webkit.org/code-style-guidelines/#names-out-argument.</a>

Canonical link: <a href="https://commits.webkit.org/293298@main">https://commits.webkit.org/293298@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7e588aa745ceffb2049d998fcd5b38dbe0f27f41

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/98407 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/18038 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8269 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/103528 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/48935 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18331 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/26490 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/74909 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32080 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/101411 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/13898 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/88880 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55268 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/13678 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/48377 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/6922 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/105901 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/25495 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-Sequoia-Debug-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/83886 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/25871 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85081 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83365 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21071 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28008 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/5679 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/19149 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/25454 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/30631 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25272 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/28592 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/26847 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/UIProcess/PDF/WKPDFHUDView.mm:
```diff
@@ -348,11 +348,11 @@ - (NSImage *)_getImageForControlName:(NSString *)control
-    _cachedIcons.get()[control] = iconImage;
-    return iconImage;
+    _cachedIcons.get()[control] = iconImage.get();
+    return iconImage.get();
```
I think this should call `autorelease()`

aprotyas replied:
Indeed, `_getImageForControlName:` does not return a +1 object so we need to autorelease it. Good catch, thanks

aprotyas replied:
Fixed in https://github.com/WebKit/WebKit/commit/b7bb4d9e7b82a9b7124a7cbad87dd584ec16982b

cdumez commented on Source/WebKit/UIProcess/PDF/WKPDFHUDView.mm:
```diff
@@ -328,16 +328,16 @@ - (void)_setupLayer:(CALayer *)parentLayer
+    RetainPtr<NSImage> iconImage = _cachedIcons.get()[control];
     if (iconImage)
-        return iconImage;
+        return iconImage.get();
```
Missing `autorelease()` here too

aprotyas replied:
My bad, correct. Fixing

whsieh replied:
FWIW, I think `get()` is technically safe here since `_cachedIcons` is guaranteed to hold on to the image as well — but `autorelease()` in the return value does seem more robust to future changes.

aprotyas replied:
Agreed, changed to `autorelease()` in https://github.com/WebKit/WebKit/commit/7e588aa745ceffb2049d998fcd5b38dbe0f27f41 though

hortont424 commented on Source/WebKit/UIProcess/PDF/WKPDFHUDView.mm:
```diff
@@ -328,16 +328,16 @@ - (void)_setupLayer:(CALayer *)parentLayer
+    [self _setupLayer:parentLayer.get()];
 }
 
 - (NSImage *)_getImageForControlName:(NSString *)control
```
also shouldn't be "get"

aprotyas replied:
Renamed!

whsieh reviewed (approved):
r=me with Chris' comment

aprotyas committed:
[Safer CPP] Address UnretainedLocalVars warnings in WKPDFHUDView.mm
https://bugs.webkit.org/show_bug.cgi?id=291077
rdar://148588361

Reviewed by Megan Gardner, Chris Dumez, and Wenson Hsieh.

This patch addresses the aforementioned class of Safer CPP warnings by
introducing more RetainPtr adoption within WKPDFHUDView.mm.

* Source/WebKit/SaferCPPExpectations/UnretainedLocalVarsCheckerExpectations:
* Source/WebKit/UIProcess/PDF/WKPDFHUDView.mm:
(-[WKPDFHUDView handleMouseUp:]):
(-[WKPDFHUDView _controlIndexForEvent:]):
(-[WKPDFHUDView _loadIconImages]):
(-[WKPDFHUDView _setupLayer:]):
(-[WKPDFHUDView _redrawLayer]):
(-[WKPDFHUDView _imageForControlName:]):
(-[WKPDFHUDView _getImageForControlName:]): Deleted.

Drive-by fix: Renamed since it does not return through out arguments --
align with https://webkit.org/code-style-guidelines/#names-out-argument.

Canonical link: https://commits.webkit.org/293298@main
